### ISSUE TITLE
Taskcluster checkout step can fail due to "Couldn't find remote ref"

### ISSUE THREAD

**Issue by foolip** [web-platform-tests/wpt#14618]
In https://tools.taskcluster.net/groups/bMtY44xAQSaHHLG9YfXPvA (for https://github.com/web-platform-tests/wpt/pull/13886) all the tasks failed like this:

```
[taskcluster 2018-12-20 12:16:53.260Z] Task ID: APDq-Cd_Rm6I7q2y5FBUyQ
[taskcluster 2018-12-20 12:16:53.260Z] Worker ID: i-0bf1fdc7a6f8d47e9
[taskcluster 2018-12-20 12:16:53.260Z] Worker Group: us-east-1
[taskcluster 2018-12-20 12:16:53.260Z] Worker Node Type: r3.xlarge
[taskcluster 2018-12-20 12:16:53.260Z] Worker Type: wpt-docker-worker
[taskcluster 2018-12-20 12:16:53.260Z] Public IP: 54.81.165.37

[taskcluster 2018-12-20 12:16:54.282Z] === Task Starting ===
+ /home/test/start.sh https://github.com/web-platform-tests/wpt.git refs/pull/13886/merge FETCH_HEAD chrome dev
++ REMOTE=https://github.com/web-platform-tests/wpt.git
++ REF=refs/pull/13886/merge
++ REVISION=FETCH_HEAD
++ BROWSER=chrome
++ CHANNEL=dev
++ cd /home/test
++ mkdir web-platform-tests
++ cd web-platform-tests
++ git init
Initialized empty Git repository in /home/test/web-platform-tests/.git/
++ git remote add origin https://github.com/web-platform-tests/wpt.git
++ git fetch --quiet --depth=50 --tags origin refs/pull/13886/merge
fatal: Couldn't find remote ref refs/pull/13886/merge
[taskcluster 2018-12-20 12:16:55.405Z] === Task Finished ===
[taskcluster 2018-12-20 12:16:55.477Z] Unsuccessful task run with exit code: 128 completed in 2.217 seconds
```

Feching refs/pull/13886/merge works for me and there are no merge conflicts, so it's not clear why this happened. I had just closed and reopened the PR, probably there's a race here. @jgraham is this race familiar to you?

Some retry logic could fix this and https://github.com/web-platform-tests/wpt/issues/14617.

**Comment by Hexcles** [comment]
This is most likely a race.

GitHub never promises the timing of the merge commit (the whole `refs/pull/*/merge` thing is undocumented). It might be temporarily unavailable or out-of-date when the PR branch is updated. I've seen this a few times, usually when a PR is updated in quick concession and the check for an earlier commit starts right when a new commit is pushed. It's not a big issue otherwise, because there is usually significant delay between the commit is pushed and the checkout on CI starts (acquiring and setting up the VM, etc.).

Note that we cannot put an infinite retry here. The merge commit may never be available if there are merge conflicts.

Race conditions are kind of unavoidable if we want the "merge into the head of master" behaviour, and there is no better solution than this undocumented ref AFAIK. Quite a lot of CIs are doing this and suffering the same issue.

In short, let's just retry a few times and live with it ;)

**Comment by foolip** [comment]
Must we use the merge ref at all? Why not just check out the PR and merge in master? In the case of merge conflicts, it'd fail in an obvious way.

Retrying with exponential back off until the job times out seems fine?

**Comment by foolip** [comment]
Well, this didn't keep happening.
